pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -343,3 +346,40 @@ func (r *Cluster) ValidateDelete() error {
 
 	return nil
 }
+
+func validateServerVersionSkew(new, old *Cluster) field.ErrorList {
```

Comment:
Can't this be reused in the CLI as well? it only takes two clusters, that's independent from the webhook

Reply:
It seems that we still have the parsing logic duplicated

Reply:
the parsing logic for getting the old server version in the CLI is using a kubectl call to get older version. with the webhooks however we can avoid this call because we have the cluster objects available.